pointers/pointers.c

```diff
@@ -66,7 +96,24 @@ int string_compare(char *m, char *n)
 */
 char *find_string(char *haystack, char *needle)
 {
-
+    unsigned int counter = 1;
+    char *hay_copy = haystack;
+    while (*hay_copy != '\0') {
+        
+        if (*hay_copy == *needle) {
+            for(unsigned int i = 1; i < strlen(needle); i++) {
+                if (*(hay_copy + i) != needle[i]) {
+                    break;
```

Comment:
Well done on this! This is a very difficult problem, and you have implemented some pretty good use cases of pointers here.